src/test/net_peer_eviction_tests.cpp

```diff
@@ -36,7 +32,7 @@ std::vector<NodeEvictionCandidate> GetRandomNodeEvictionCandidates(const int n_c
             /* nKeyedNetGroup */ random_context.randrange(100),
             /* prefer_evict */ random_context.randbool(),
             /* m_is_local */ random_context.randbool(),
-            /* m_is_onion */ random_context.randbool(),
+            /* m_network */ static_cast<Network>(random_context.randrange(static_cast<int>(Network::NET_INTERNAL))),
```

Comment:
Good point, looks like your suggestion with this might do it:
```diff
diff --git a/src/test/net_peer_eviction_tests.cpp b/src/test/net_peer_eviction_tests.cpp
index ca3f943123..f827230e1a 100644
--- a/src/test/net_peer_eviction_tests.cpp
+++ b/src/test/net_peer_eviction_tests.cpp
@@ -4,6 +4,7 @@
 
 #include <netaddress.h>
 #include <net.h>
+#include <test/util/net.h>
 #include <test/util/setup_common.h>
 
 #include <boost/test/unit_test.hpp>
@@ -32,7 +33,7 @@ std::vector<NodeEvictionCandidate> GetRandomNodeEvictionCandidates(const int n_c
             /* nKeyedNetGroup */ random_context.randrange(100),
             /* prefer_evict */ random_context.randbool(),
             /* m_is_local */ random_context.randbool(),
-            /* m_network */ static_cast<Network>(random_context.randrange(static_cast<int>(Network::NET_INTERNAL))),
+            /* m_network */ ALL_NETWORKS[random_context.randrange(ALL_NETWORKS.size())],
         });
     }
     return candidates;
diff --git a/src/test/util/net.h b/src/test/util/net.h
index 4e7c446af1..5f3f412cc9 100644
--- a/src/test/util/net.h
+++ b/src/test/util/net.h
@@ -68,7 +68,7 @@ constexpr ConnectionType ALL_CONNECTION_TYPES[]{
     ConnectionType::ADDR_FETCH,
 };
 
-constexpr Network ALL_NETWORKS[]{
+constexpr std::array ALL_NETWORKS{
     Network::NET_UNROUTABLE,
     Network::NET_IPV4,
     Network::NET_IPV6,
```

Reply:
Done.

Reply:
Huh, the CI is barfing on this change. Strange.

Reply:
This might be a compiler bug. I used `static constexpr auto ALL_FEE_ESTIMATE_HORIZONS = std::array{` to work around that.

Reply:
Thanks! Will try that.

Reply:
It worked 👍